gossip/gossip.go

```diff
@@ -691,21 +696,22 @@ func (g *Gossip) hasOutgoing(nodeID roachpb.NodeID) bool {
 func (g *Gossip) getNextBootstrapAddress() net.Addr {
 	// Run through resolvers round robin starting at last resolved index.
 	for i := 0; i < len(g.resolvers); i++ {
-		g.resolverIdx = (g.resolverIdx + 1) % len(g.resolvers)
+		g.resolverIdx++
+		g.resolverIdx %= len(g.resolvers)
 		g.resolversTried[g.resolverIdx] = struct{}{}
 		resolver := g.resolvers[g.resolverIdx]
-		addr, err := resolver.GetAddress()
-		if err != nil {
-			log.Errorf("invalid bootstrap address: %+v, %v", resolver, err)
-			continue
-		} else if addr.String() == g.is.NodeAddr.String() {
-			// Skip our own node address.
+		if resolver.IsExhausted() {
```

Comment:
This line is basically irrelevant - the exhaustedness logic is pretty much completely unhooked from all non-test code. See https://github.com/cockroachdb/cockroach/issues/5801